python_modules/dagster/dagster/_core/definitions/decorators/source_asset_decorator.py

```diff
@@ -0,0 +1,118 @@
+from typing import TYPE_CHECKING, Mapping, Optional, Sequence, Union, overload
+
+from dagster._core.definitions.events import AssetKey, CoercibleToAssetKeyPrefix
+from dagster._core.definitions.metadata import (
+    MetadataEntry,
+    MetadataUserInput,
+    PartitionMetadataEntry,
+)
+from dagster._core.definitions.partition import PartitionsDefinition
+from dagster._core.definitions.resource_definition import ResourceDefinition
+from dagster._core.definitions.source_asset import SourceAsset, SourceAssetObserveFunction
+from dagster._core.storage.io_manager import IOManagerDefinition
+
+from ..decorators.op_decorator import _Op
+
+if TYPE_CHECKING:
+    from dagster._core.execution.context.compute import SourceAssetObserveContext
+
+
+@overload
+def observable_source_asset(observe_fn: SourceAssetObserveFunction) -> SourceAsset:
+    ...
+
+
+@overload
+def observable_source_asset(
+    *,
+    name: Optional[str] = ...,
+    key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    metadata: Optional[MetadataUserInput] = None,
+    io_manager_key: Optional[str] = None,
+    io_manager_def: Optional[IOManagerDefinition] = None,
+    description: Optional[str] = None,
+    partitions_def: Optional[PartitionsDefinition] = None,
+    _metadata_entries: Optional[Sequence[Union[MetadataEntry, PartitionMetadataEntry]]] = None,
+    group_name: Optional[str] = None,
+    resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
+) -> "_ObservableSourceAsset":
+    ...
+
+
+def observable_source_asset(
```

Comment:
I think it would be safest to mark this as experimental for now, though open to otherwise if you disagree

Reply:
agreed